### PR TITLE
Added search and recent videos, as well as authentication

### PR DESCRIPTION
**Pull request by mongkuen**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class UsersController < ApplicationController
+  before_action :redirect_to_home
+
+  def new
+    @user = User.new
```
Comment (alagram) [inline app/controllers/users_controller.rb:5]: Ideally this action should not be accessed by logged in users. You can do `redirect_to some_path and return if logged_in?`

File: app/models/category.rb [hunk 2]
```diff
@@ -1,3 +1,8 @@
 class Category < ActiveRecord::Base
-  has_many :videos, -> { order("title") }
+  has_many :videos, -> { order(created_at: :desc) }
+
+  def recent_videos
+    self.videos.limit(6)
```
Comment (alagram) [inline app/models/category.rb:5]: You can omit `self` since it's implied.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (mongkuen) [inline app/controllers/users_controller.rb:5]: I believe the `before_action` is the method that handled that. It redirects to the home path if the user is logged in.

File: app/models/category.rb (hunk 2 above)
Comment (mongkuen) [inline app/models/category.rb:5]: I use it as a visual reminder that's I'm acting on a instance. Is it in bad taste to explicitly declare it?

Thank you so much for your feedback Albert!

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (alagram) [inline app/controllers/users_controller.rb:5]: Cool!

File: app/models/category.rb (hunk 2 above)
Comment (alagram) [inline app/models/category.rb:5]: Nope. Not really, it's tiny nitpick.

**Pull request merged**